sw/airborne/modules/ctc/ctc.h

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2017  Hector Garcia de Marina
+ *
+ * This file is part of paparazzi.
+ *
+ * paparazzi is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2, or (at your option)
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ *
+ */
+
+/** @file ctc.h
+ *
+ *  Collective Tracking Control
+ */
+
+#ifndef CTC_H
+#define CTC_H
+
+#include "std.h"
+
+typedef struct {
+  float k1;
+  float k2;
+  float alpha;
+  uint16_t timeout;
+  float p_centroid_x;
```

Comment:
don't forget that you can use the `struct FloatVect2` and corresponding math operations from `math/pprz_algebra.h` and `math/pprz_algebra_float.h` to have a more compact code.

Reply:
I totally forgot about it. I would prefer to do not touch this part of the code since it is working (validated).

I have another algorithm to implement later. I will use these libraries for sure.